### ISSUE TITLE
Header and delete button

### ISSUE THREAD

**Issue by DmitrySkripkin** [quilljs/quill#1848]
This behavior is different from all text editors. You can try how it works in google docs or dropbox paper, notion and etc. I fixed it with specific keyboard binding for right now, but I think it can be bigger issue if I dig deeper.

**Steps for Reproduction**

1. Visit [quilljs.com](https://quilljs.com/)
2. Set caret after header
3. Hit delete button

**Expected behavior**:
Next paragraph is appended  to header

**Actual behavior**:
Header is prepended to next paragraph

**Platforms**:
Any

**Version**:
Any

**Comment by DmitrySkripkin** [comment]
I think it's the same issue. Instead "Hit delete button" paste multiple paragraphs. The last one will have title format. And title is a plain text now.

**Comment by thomsbg** [comment]
I just ran into this. Some more details:

When using the Delete key (aka forward delete) at the end of a line, the formatting of the original line is lost, with the formatting of the next line taking precedence. Using the backspace key at the beginning of the next line has the correct behavior.

**Steps for Reproduction**

1. Visit quilljs.com
2. Place cursor at the end of the first line, after the `r` in `Editor`
3. Press the Delete key (fn + Delete on MacOS).

**Expected behavior**:

The line below the heading is removed, and the first line keeps all its formatting.

**Actual behavior**:

The first line is stripped of its formats, becoming left-aligned and normal size.

**Platforms**:

Tested on Chrome 65.0.3325.181, MacOS

**Version**:

1.3.6

**More details**:

This bug also happens when using a range delete to merge lines, where the range ends at the beginning of a line:

1. Visit quilljs.com
2. Place cursor at the end of the first line
3. Use shift+right to expand selection to beginning of the second line
4. Press delete (or backspace, doesn't matter which)

See the first line lose its formatting unexpectedly. Oddly enough, using cmd+x to cut in this situation has the correct behavior.
